engine/access/rpc/engine.go

```diff
@@ -161,16 +162,14 @@ func NewBuilder(log zerolog.Logger,
 		}
 	}
 
-	connectionFactory := &backend.ConnectionFactoryImpl{
+	connectionFactory := &connection.ConnectionFactoryImpl{
 		CollectionGRPCPort:        collectionGRPCPort,
 		ExecutionGRPCPort:         executionGRPCPort,
 		CollectionNodeGRPCTimeout: config.CollectionClientTimeout,
 		ExecutionNodeGRPCTimeout:  config.ExecutionClientTimeout,
-		ConnectionsCache:          cache,
-		CacheSize:                 cacheSize,
-		MaxMsgSize:                config.MaxMsgSize,
 		AccessMetrics:             accessMetrics,
 		Log:                       log,
+		Manager:                   connection.NewManager(connection.NewCache(cache, int(cacheSize)), log, accessMetrics, config.MaxMsgSize),
```

Comment:
`cache` may be nil, in which case we should pass nil instead of `connection.NewCache(cache, int(cacheSize))`.